clients/java/signalr/src/main/java/HubConnection.java

```diff
@@ -15,21 +15,38 @@
     private CallbackMap handlers = new CallbackMap();
     private HubProtocol protocol;
     private Gson gson = new Gson();
+    private Boolean handshakeReceived = false;
+    private static final String RECORD_SEPARATOR = "\u001e";
     private HubConnectionState connectionState = HubConnectionState.DISCONNECTED;
 
+    // Only json is supported currently
+    private String protocolName = "json";
+    private int protocolVersion = 1;
+
     public HubConnection(String url, Transport transport) {
         this.url = url;
         this.protocol = new JsonHubProtocol();
         this.callback = (payload) -> {
 
+            if (!handshakeReceived) {
+                int handshakeLength = payload.indexOf(RECORD_SEPARATOR) + 1;
+                payload = payload.substring(handshakeLength);
+                handshakeReceived = true;
+
+                // We only received the handshake response ,so we can return.
+                if (payload.length() == 0){
+                    return;
+                }
+            }
+
             HubMessage[] messages = protocol.parseMessages(payload);
 
             for (HubMessage message : messages) {
                 switch (message.getMessageType()) {
                     case INVOCATION:
                         InvocationMessage invocationMessage = (InvocationMessage)message;
                         if (message != null && handlers.containsKey(invocationMessage.target)) {
-                            ArrayList<Object> args = gson.fromJson((JsonArray)invocationMessage.arguments[0], (new ArrayList<Object>()).getClass());
+                            ArrayList<Object> args = gson.fromJson((JsonArray)invocationMessage.arguments[0], (new ArrayList<>()).getClass());
```

Comment:
Curious why `new ArrayList<Object>` changed to `new ArrayList<>`. Is this intentional?

Reply:
Yeah, the `Object` on the other side is not needed so I just took it out.  Nothing to do with this change specifically

Reply:
Cool, sounds good.